File: summationOfPrimes.py
# Problem: Find the total_sum of all the primes not greater than given N.
# Solution: Time limit exceeded in these examples.

def sum_of_primes(n):
    prime = [True for num in range(n + 1)]
    p = 2
    total_sum = 0
    while (p <=n):
        if (prime[p]):
            total_sum += p
            for i in range(p * p, n + 1, p):
                prime[i] = False

        p += 1
    return total_sum

File: summationOfPrimes.js
// Problem: Find the sum of all the primes not greater than given N.
// Solution: Ues the sieve of eratosthenes and then sum the primes.

function sumOfPrimes(N) {
    let primes = new Array(N + 1).fill(true);
    primes[0] = false;
    primes[1] = false;
    let sum = 0;

    for (let p = 2; p <= N; p++) {
        if (primes[p]) {
            sum += p;
            for (let i = p*p; i <= N; i += p) {
                primes[i] = false;
            }
        }
    }
    return sum;
}
